ipatests/test_integration/test_installation.py

```diff
@@ -1364,6 +1364,36 @@ def test_kra_certs_renewal(self):
             )
             assert starting_serial != int(cert.serial_number)
 
+    def test_sidgen_task_continue_on_error(self):
+        """Verify that SIDgen task continue even if it fails to assign sid
+
+        related: https://pagure.io/freeipa/issue/9618
+        """
+        test_user = 'test_user2000'
+        # test_user_uid = 2000
+        old_err_msg = 'Cannot add SID to existing entry'
+        new_err_msg = r'Finished with [0-9]+ failures, please check the log'
+
+        tasks.kinit_admin(self.master)
+
+        self.master.run_command(
+            ['ipa', 'user-add', test_user,
+             '--first', 'test', '--last', 'user',
+             '--uid', 2000]
+        )
+
+        self.master.run_command(
+            ['ipa', 'config-mod', '--add-sids', '--enable-sid']
+        )
+
+        dashed_domain = self.master.domain.realm.replace(".", '-')
+        dirsrv_error_log = self.master.get_file_contents(
+            paths.SLAPD_INSTANCE_ERROR_LOG_TEMPLATE % (dashed_domain),
+            encoding='utf-8'
+        )
```

Comment:
Do you need to wait for the task to finish? I'm not sure you'd be able to reliably pull the task id out of a log and wait on it so may just have to go with a sleep(something). I imagine this would generally be pretty fast so maybe 5 seconds? There are only two users after all.

Reply:
updated.